### PR TITLE
Added better information for changes in the history tab of an Asset

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
This adds two methods `changedInfo()` and `bulkChangedInfo()`.

`changedInfo()` and `bulkChangedInfo()` are similar except `changedInfo()` utilizes the model scopes for new changes.
array keys have been renamed to be more readable (e.g `rtd_location_id` is now `Default Location`) in the History tab of an asset. This information will be displayed if you edit/update a single asset or multiple assets at once.

If an asset never had a supplier, default location, company, or location already assigned to it, it will be noted as 'unassigned' (example below). If supplier, default location, company, or location are removed it will go back to unassigned as well.

Before:
image

After:
image

Removed the Supplier:
image

Fixes #sc-14441

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 1]
```diff
@@ -197,6 +202,48 @@ public function update(Request $request)
...
+            unset($changed['model_id']);
+        }
+        if(array_key_exists('company_id', $changed)) {
+            $changed['company_id']['old'] = $changed['company_id']['old'] ? Company::find($changed['company_id']['old'])->name : 'unassigned';
+            $changed['company_id']['new'] = Company::find($changed['company_id']['new'])->name;
+            $changed['Company'] = $changed['company_id'];
+            unset($changed['company_id']);
+        }
+        if(array_key_exists('supplier_id', $changed)) {
+            $changed['supplier_id']['old'] = $changed['supplier_id']['old'] ? Supplier::find($changed['supplier_id']['old'])->name : 'unassigned';
```
Comment (marcusmoore) [inline app/Http/Controllers/Assets/BulkAssetsController.php]: You gotta import `Supplier`.

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 2]
```diff
@@ -197,6 +202,48 @@ public function update(Request $request)
         // no values given, nothing to update
         return redirect($bulk_back_url)->with('warning', trans('admin/hardware/message.update.nothing_updated'));
     }
+    /**
+     * This takes the ids of the changed attributes and returns the names instead for the history view of an Asset
+     *
+     * @param  $asset
+     * @return void
```
Comment (marcusmoore) [inline app/Http/Controllers/Assets/BulkAssetsController.php]: I think the type hint should be `array`.

File: app/Observers/AssetObserver.php [hunk 3]
```diff
@@ -59,6 +66,49 @@ public function updating(Asset $asset)
             $logAction->logaction('update');
         }
     }
+    /**
+     * This takes the ids of the changed attributes and returns the names instead for the history view of an Asset
+     *
+     * @param  $asset
+     * @return void
```
Comment (marcusmoore) [inline app/Observers/AssetObserver.php]: I think the type hint should be `array`.

**Review by marcusmoore — CHANGES_REQUESTED** [review]
Overall this looks good. A couple minor things (the missing Supplier import is the biggest)

**Comment by Godmartinz** [comment]
Whoops on the import, I swear it worked! 😅

**Comment by Godmartinz** [comment]
Closing this because, @spencerrlongg made a good point, that I am changing the way data is being stored in the DB and not adjusting old data. It would be better if I put this method inside of the actionlogtransformer and did my changes basically on the front end not the back end. Love the idea!

**Pull request closed without merging**